### PR TITLE
chore(helm): stop requesting superfluous permissions

### PR DESCRIPTION
**Pull request by oleobal**
## Description

Remove permissions that we think are superfluous.

(this was split from #671)

## How has this been tested?

⬇️ (sufficient, I hope)

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [x] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by SdgJlbl — APPROVED** [review]
Do we have a process to ensure that the helm chart version is the right one? 🤔

**Comment by oleobal** [comment]
@SdgJlbl 

> Do we have a process to ensure that the helm chart version is the right one? 🤔

Sort of, the CI checks the proposed version doesn't already exist on the repo: https://github.com/Substra/substra-backend/blob/main/.github/workflows/helm.yml#L41

but that's it

File: charts/substra-backend/templates/rbac.yaml [hunk 1]
```diff
@@ -20,9 +20,12 @@ rules:
     resources: ["secrets"]
     verbs: ["get", "watch", "list", "create", "update", "patch"]
```
Comment (guilhem-barthes) [inline charts/substra-backend/templates/rbac.yaml:21]: Do we actually need "create", "update", "patch" on secrets? It looks like this Role just use read access

File: charts/substra-backend/templates/rbac.yaml [hunk 2]
```diff
@@ -20,9 +20,12 @@ rules:
...
   - apiGroups: [""]
-    resources: ["pods", "pods/log", "pods/status", "pods/exec"]
+    resources: ["pods/log", "pods/status"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["pods", "pods/exec"]
     verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
-  - apiGroups: ["batch", "extensions"]
+  - apiGroups: ["batch"]
```
Comment (guilhem-barthes) [inline charts/substra-backend/templates/rbac.yaml:28]: Do we actually use batch jobs in the worker?

Comment (oleobal) [reply charts/substra-backend/templates/rbac.yaml:28]: Ctrl+F-ing `batch` in the code base would lead my to believe we don't

Comment (guilhem-barthes) [reply charts/substra-backend/templates/rbac.yaml:28]: But do we need to run jobs in the worker? I thought we just ran pods :x

Comment (SdgJlbl) [reply charts/substra-backend/templates/rbac.yaml:28]: To the best of my knowledge, we don't use K8s Jobs currently

**Pull request merged**